lib/reporter.ex

```diff
@@ -0,0 +1,143 @@
+defmodule Membrane.TelemetryMetrics.Reporter do
+  @moduledoc """
+  Attaches handlers to :telemetry events basing on received list of metrics definitions.
+  Attached handlers store metrics values in ETS tables.
+  These values can be get by calling `scrape/2` function or also reseted by calling `scrape_and_cleanup/2`.
+
+  Currently supported types of metrics are:
+   - `Telemetry.Metrics.Counter`
+   - `Telemetry.Metrics.Sum`
+   - `Telemetry.Metrics.LastValue`
+
+  Currently supported fields of metrics definitions are: `:name`, `:event_name`, `measurement`.
+  Fields `:keep`, `:reporter_options`, `tag_values`, `tags`, `:unit` and functionalities related to them are not supported yet.
+  Metrics values are grouped by  value related to key `:telemetry_metadata` in event `metadata`.
+  """
+
+  use GenServer
+
+  alias Membrane.TelemetryMetrics.Reporter.{Counter, LastValue, Sum}
+
+  @type reporter() :: pid() | atom()
+  @type report() :: map()
+
+  @spec start_link(list(), GenServer.options()) :: GenServer.on_start()
```

Comment:
This is specifically a keyword list, right?

```suggestion
  @spec start_link(Keyword.t(), GenServer.options()) :: GenServer.on_start()
```